turbopuffer/namespace.py

```diff
@@ -319,10 +319,18 @@ def upsert(self, data=None, ids=None, vectors=None, attributes=None, schema=None
         else:
             raise ValueError(f'Unsupported data type: {type(data)}')
 
-    def delete(self, ids: Union[int, str, List[int], List[str]]) -> None:
+    def delete(self, ids: Union[int, str, List[int], List[str], List[Union[int, str]]]) -> None:
         """
         Deletes vectors by id.
         """
+        # If it's list[int | str], we need to convert to list[int] and list[str]
+        if isinstance(ids, list):
+            if all(isinstance(id, int) for id in ids):
+                ids = [int(id) for id in ids]  # type: List[int]
+            elif all(isinstance(id, str) for id in ids):
+                ids = [str(id) for id in ids]  # type: List[str] 
```

Comment:
this does not look free

how do we end up with this `list[int | str]` type? namespaces should either have int or str ids

and even if we do, why do we need to convert the type in the client? JSON can serialize fine